lib/block_view.go

```diff
@@ -3807,17 +3754,30 @@ func (bav *UtxoView) GetDiamondEntryForDiamondKey(diamondKey *DiamondKey) *Diamo
 	// If we get here it means no value exists in our in-memory map. In this case,
 	// defer to the db. If a mapping exists in the db, return it. If not, return
 	// nil.
-	dbDiamondEntry := DbGetDiamondMappings(
-		bav.Handle, &diamondKey.ReceiverPKID, &diamondKey.SenderPKID, &diamondKey.DiamondPostHash)
-	if dbDiamondEntry != nil {
-		bav._setDiamondEntryMappings(dbDiamondEntry)
+	var diamondEntry *DiamondEntry
+	if bav.Postgres != nil {
+		diamond := bav.Postgres.GetDiamond(&diamondKey.SenderPKID, &diamondKey.ReceiverPKID, &diamondKey.DiamondPostHash)
```

Comment:
Nit: It feels like it would be kinda nice if the raw Postgres querying, and the raw manipulation of PGDiamond and other PG* structs was encapsulated into postgres.go rather than bubbled up here. That would give a clean interface that would allow us to potentially swap out the storage layer in the future. That said, if there are many instances of this to the point where wrapping these things in their own functions is annoying then I would skip it.

Reply:
On second thought just don't worry about this. It's too annoying and not needed.